toolchain/lower/handle_type.cpp

```diff
@@ -7,64 +7,22 @@
 
 namespace Carbon::Lower {
 
-auto HandleArrayType(FunctionContext& context, SemIR::InstId inst_id,
-                     SemIR::ArrayType /*inst*/) -> void {
-  context.SetLocal(inst_id, context.GetTypeAsValue());
-}
-
-auto HandleAssociatedEntityType(FunctionContext& context, SemIR::InstId inst_id,
-                                SemIR::AssociatedEntityType /*inst*/) -> void {
-  context.SetLocal(inst_id, context.GetTypeAsValue());
-}
-
-auto HandleClassType(FunctionContext& context, SemIR::InstId inst_id,
-                     SemIR::ClassType /*inst*/) -> void {
-  context.SetLocal(inst_id, context.GetTypeAsValue());
-}
-
-auto HandleConstType(FunctionContext& context, SemIR::InstId inst_id,
-                     SemIR::ConstType /*inst*/) -> void {
-  context.SetLocal(inst_id, context.GetTypeAsValue());
-}
+// For potentially non-constant instructions that are always of type `type`
+// and might define a type constant, produce the trivial runtime representation
+// of type `type`.
```

Comment:
FWIW, the "potentially" here throws me a little. Would the below suggestion carry similar meaning for you, trying to say similar with less?

```suggestion
// For instructions that are always of type `type`, produce the trivial runtime representation
// of type `type`.
```

Reply:
I'm fine with this change. For me the new comment doesn't explain why we filter out the `CONSTANT_ALWAYS` case below, but perhaps that's obvious enough that it's not worth calling out.